validation/src/main/scala/hmda/validation/api/ValidationApi.scala

```diff
@@ -0,0 +1,24 @@
+package hmda.validation.api
+
+import hmda.validation.dsl.{ Failure, Result, Success }
+import hmda.validation.engine.ValidationError
+import scalaz._
+import scalaz.Scalaz._
+
+trait ValidationApi {
+
+  def convertResult[A](input: A, result: Result, ruleName: String): ValidationNel[ValidationError, A] = {
```

Comment:
A is just a generic type, usually called `type parameters` in Scala. So the signature of this method is saying something along the lines of "for a given input of type `A`, a `Result` and a name, return a `ValidationError` with type parameters `ValidationError` and `A`"

A lot of Scala code uses `T` for generic types, as is done in Java. Some Scala naming conventions prefer to use a capital letter of the English alphabet, starting with `A`. I use `T` (as in "Type") when it's the only type parameter of a function, i.e. 

```scala
def function[T] (i: Int, s: String): T = ???
```
It there is more than one, I usually start with `A`, `B` (but it could also be `K`, `V` if you are doing keys and values for instance), etc. i.e.:

```scala
def function (a: A, b: B) = ???
```

This is all based on convention, we can use one or the other but I agree it should be consistent. Since there are more `T`s than `A`s in the codebase, if it's ok with you I will change the convertResult signature to use a `T`